tests/compile-fail/non_extern_foreign_function_clashing.rs

```diff
@@ -0,0 +1,20 @@
+// revisions: two three
+
+#[no_mangle]
+fn foo() {}
+
+#[export_name = "foo"]
+fn bar() {}
+
+#[cfg(three)]
+#[export_name = "foo"]
+fn baz() {}
```

Comment:
I want to test my `NOTE` here, but it then starts to complain about a lot of unexpected "inside ..." notes. Is it possible to test these notes?

Reply:
> (I also removed revisions because I really don't want to duplicate the long error messages and I don't know a way to share them between revisions.)

Which long error messages?

Reply:
The combination of the two `HELP`s and the `ERROR` feels a little long to me. But maybe they are fine?

Reply:
It's not like we are changing this code all the time so I don't see the problem.
Also, if you have comments like `//~ ERROR foo` (the usual error comments) they apply to all revisions, and you can even do `//[rev1,rev2]~ ERROR` to have a message in 2 but not all revisions. I don't know if that's what you mean by "share" though as I still have not understood what it is you are not doing here.

Reply:
> you can even do `//[rev1,rev2]~ ERROR` to have a message in 2 but not all revisions

That does not seem to work...

I was trying to test that both two and three clashing symbols would trigger the error. Without `revisions`, this test currently only test that exactly two symbols would trigger the error. Maybe that's enough?

Reply:
> That does not seem to work...

Weird, I've used this in the past.... but maybe this feature has not been ported from rustc's conmpiletest to the standalone crate.

But yeah, what you describe sounds enough.